platform/android/src/style/sources/sources.cpp

```diff
@@ -20,7 +20,9 @@ namespace {
 
     Source* initializeSourcePeer(mbgl::style::Source& coreSource) {
         Source* source;
-        if (coreSource.is<mbgl::style::VectorSource>()) {
+        if (coreSource.peer.has_value()) {
```

Comment:
I think this case should be checked by `createJavaSourcePeer`, not by `initializeSourcePeer`. The return type of `initializeSourcePeer` should be `std::unique_ptr<Source>`.

Reply:
`initializeSourcePeer` isn't just an initializer anymore. I think it can be changed to `getSourcePeer` and skip the `unique_ptr<>` since it's not really needed in `createJavaSourcePeer` or `removeSourceFromMap`